core/pipeline.py

```
@@ -1,76 +1,206 @@
+        
+        # Add user input to conversation history
+        self.conversation_history.append({"role": "user", "content": user_input})
+
+        # Check if we need to generate a new summary (every 10 messages)
+        if len(self.conversation_history) % 10 == 0:
+            summary = generate_conversation_summary(self.conversation_history, self.user_id)
+            save_conversation_summary(self.user_id, summary)
+            self.last_summary_update = datetime.utcnow().isoformat()
+
+        # Get existing conversation summary
+        conversation_summary = get_conversation_summary(self.user_id)
```

Ruff (0.8.2)

Local variable `conversation_summary` is assigned to but never used

Remove assignment to unused variable `conversation_summary`

(F841)